### ISSUE TITLE
Subscription never ready

### ISSUE THREAD

**Issue by samcorcos** [meteor/react-packages#79]
I've followed the example from [here](http://react-in-meteor.readthedocs.org/en/latest/meteor-data/) almost exactly, but my `.ready()` never returns true. 

I tried making a minimum viable bug in MeteorPad, but there are too many things going on. The only way this would work is if you could clone the repo [here](https://github.com/samcorcos/meteor-react-ionic-chat) :(

The relevant files are `client/components/other.jsx`, [lines 20-25](https://github.com/samcorcos/meteor-react-ionic-chat/blob/master/client/components/Other.jsx#L20-L25), and `server/server.jsx` [lines 25-27](https://github.com/samcorcos/meteor-react-ionic-chat/blob/master/server/server.jsx#L25-L27). 

The data should be available when a user goes to the `/other` route and clicks on the "new messages" icon in the top right. This launches a modal which is populated with the `NewConversationModal` component. 

I tried accessing the data outside the modal but it does not seem to work. That said, when I log the query from the server or from a `Meteor.method`, everything seems to be in order. 

Do let me know if you have any questions! I'd really like to figure out what's going wrong here.

**Comment by stubailo** [comment]
@samcorcos that's interesting; I feel like it should work. We'll look into it. Thanks for providing a whole app reproduction!

**Comment by samcorcos** [comment]
@stubailo Thanks! As always, I appreciate your responsiveness :)

**Comment by samcorcos** [comment]
Any luck with this? I started work on a second app and ran into the same problem :(

**Comment by giantelk** [comment]
A few things, the users collection wasn't showing on the client, adding `meteor add autopublish` now shows the users collection. So try adding a `subscription` on the client.

Also, your code was putting JSX/HTML in this.state. That was really hard to follow. I refactored, see my repo here. Although there's a bug where the modal isn't showing. If you go with my pattern, you may want to bring the this.state.isShowModal up a level if you wanted to reuse the modal code in other screens.
https://github.com/giantelk/meteor-react-ionic-chat

It would be really helpful for issues like this to remove all the non essentials to the bug, i.e. remove all ionic and modal stuff. Would make it easier & faster to debug.

Either way this doesn't appear to be a react or meteor issue. btw If you use my code or want a Pull Request, let me know why the modal isn't opening -) I spend a few hours messing with this, so hopefully this much helps. -)

**Comment by giantelk** [comment]
Also, your code as is from your repo wouldn't work until I did `meteor update`.

**Comment by samcorcos** [comment]
@giantelk Wow, that's a lot of refactoring! Thanks for putting in so much time :) I'm just finishing the series on the Tinder clone, then I'll jump back into the chat app—probably later today or Sunday at the latest.
